apps/src/code-studio/header.js

```diff
@@ -99,83 +91,33 @@ header.build = function(
     stageExtrasEnabled
   );
 
-  /**
-   * Track boolean "visible" state of header popup to avoid
-   * expensive lookup on window resize.
-   * @type {boolean}
-   */
-  var isHeaderPopupVisible = false;
-
-  function showHeaderPopup() {
-    firehoseClient.putRecord(
-      {
-        study: 'mini_view',
-        event: 'mini_view_opened',
-        data_json: JSON.stringify({
-          current_level_id: currentLevelId
-        })
-      },
-      {includeUserId: true}
-    );
-    sizeHeaderPopupToViewport();
-    $('.header_popup').show();
-    $('.header_popup_link_glyph').html('&#x25B2;');
-    $('.header_popup_link_text').text(msg.lessAllCaps());
-    $(document).on('click', hideHeaderPopup);
-    progress.renderMiniView(
-      $('.user-stats-block')[0],
-      scriptName,
-      currentLevelId,
-      progressData.linesOfCodeText,
-      scriptData.student_detail_progress_view
+  // Hold off on rendering HeaderMiddle.  This will allow the "app load"
+  // to potentially begin before we first render HeaderMiddle, giving HeaderMiddle
+  // the opportunity to wait until the app is loaded before rendering.
+  $(document).ready(function() {
+    ReactDOM.render(
```

Comment:
Rather than using jsx syntax in a js file, create this with createElement. Details here: https://github.com/code-dot-org/code-dot-org/pull/35351#discussion_r441805125

Reply:
@Hamms @jmkulwik I would suggest that we *do* and *should continue to* allow JSX syntax in .js files.

We certainly _can_ safely use jsx syntax in js files, the way our system is configured today. And we certainly _do_ it frequently today:
![Screen Shot 2020-07-08 at 2 19 16 PM](https://user-images.githubusercontent.com/8001765/86971475-18f74e00-c126-11ea-93d7-65d9d250f7d1.png)

If we did decide at some point to not to allow jsx syntax in js files, then I'd be strongly in favor of renaming the file, rather than avoiding the syntax, so that people are not using two different coding styles based on the somewhat arbitrary decision of what extension the file happened to be given initially.

As for whether we _should_ allow JSX syntax in .js files (rather than requiring renaming the file), here is one popular opinion on the internet for why to allow it: https://github.com/airbnb/javascript/pull/985#issuecomment-239145468

1. because of convenience:
> If you are already using something like JSX, keeping JS separate from JSX hurts ergonomics in my experience. It feels silly to have to rename a single file just because you need a single line of JSX there.

2. (paraphrasing) because the idea of what Javascript is at any given moment in time is already totally fluid, based on what set of language extensions we have enabled at a given point in time: https://github.com/code-dot-org/code-dot-org/blob/c095d64a5e0d580a25754eb82058eb237579ccfb/apps/babel.config.json#L3-L14 and whether react syntax is enabled is not substantially different from whether those other language extensions are enabled.

I'm not trying to lay down the law here... I want this to be a discussion... and there are a number of possible outcomes. But do want to reach some agreement and get out of a situation where we have different unwritten rules being recommended by different team members. CC @islemaster for any opinions too.

Reply:
Perhaps more concretely, from the first few lines of the style guide, which I think we are still following: https://github.com/airbnb/javascript/tree/master/react

![Screen Shot 2020-07-08 at 2 47 30 PM](https://user-images.githubusercontent.com/8001765/86973783-fa935180-c129-11ea-82ea-08c104e53b84.png)

Reply:
I'm with Dave on this one.  JSX is preferred everywhere.

I'm not aware of any ways in which our current build process treats .js and .jsx files differently.  Our [babel-loader configuration](https://github.com/code-dot-org/code-dot-org/blob/c6f2dd801c76cbd733d9c98fe84b003c75fa1cae/apps/webpack.js#L133-L146) treats them exactly the same.  If anything, I wouldn't mind renaming all of our .jsx files to .js to drop any pretense that they are different.

Reply:
I will go back to JSX.  I had assumed this was a linter-related issue, but if it's fine to lint JSX in a .js file, I agree that it's better to use one coding style rather than two.

Reply:
I'm in favor of consistency; I would prefer to keep the syntaxes separate, but if we're committed to mixing them then I agree we should drop the `.jsx` extension.